cunumeric/module.py

```diff
@@ -3446,6 +3446,38 @@ def diagonal(
     )
 
 
+@add_boilerplate("a", "indices", "values")
+def put(
+    a: ndarray, indices: ndarray, values: ndarray, mode: str = "raise"
+) -> None:
+    """
+    Set storage-indexed locations to corresponding values.
```

Comment:
I find this a little cryptic. It also doesn't make explicit that the indices act as if the `a` array was flattened. Could we do something like:

```
Replaces specified elements of an array with given values.

The indexing works as if the target array had been flattened.
```